File: frontend/src/types/salesOrder.ts
export interface SalesOrder {
  id: number
  customer_id: number | null
  customer_name?: string
  order_date: string
  status: string
  total_amount: number | null
  user_id: number
}

export interface OrderItem {
  id: number
  order_id: number
  product_id: number
  quantity: number
  unit_price: number
}

export interface OrderItemInput {
  productId: number
  quantity: number
}

File: worker/internal/models/sales_order.go
package models

import "time"

// SalesOrderPDFRequest representa el mensaje para generar un PDF de orden de venta
type SalesOrderPDFRequest struct {
	OrderID int64  `json:"order_id"`
	UserID  int64  `json:"user_id"`
	Email   string `json:"email"`
}

// SalesOrderWithItems contiene todos los datos necesarios para el PDF
type SalesOrderWithItems struct {
	ID           int64          `json:"id"`
	CustomerName string         `json:"customer_name"`
	OrderDate    time.Time      `json:"order_date"`
	Status       string         `json:"status"`
	TotalAmount  float64        `json:"total_amount"`
	Items        []OrderItemPDF `json:"items"`
}

// OrderItemPDF representa un item de la orden para el PDF
type OrderItemPDF struct {
	ProductName string  `json:"product_name"`
	Quantity    int     `json:"quantity"`
	UnitPrice   float64 `json:"unit_price"`
	Subtotal    float64 `json:"subtotal"`
}
